File: 128.longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        sett = set(nums)

        longest = 0

        for num in sett:

            if num - 1 not in sett:

                current_streak = 1
                current_num = num

                while current_num + 1 in sett:
                    current_streak += 1
                    current_num += 1
                
                longest = max(longest, current_streak)
            
        return longest

File: 128.longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();

        for (int num : nums) {
            set.add(num);
        }

        int longest = 0;

        for (int num : set) {
            if (!set.contains(num - 1)) {
                int current = num;
                int streak = 1;

                while (set.contains(current + 1)) {
                    current += 1;
                    streak += 1;
                }

                longest = Math.max(longest, streak);
            }
        }

        return longest;
    }
}